File: server/src/main/java/com/loan_calculator/api/domain/loan/Loan.java
package com.loan_calculator.api.domain.loan;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import com.loan_calculator.api.domain.customer.Customer;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "loan")
public class Loan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime data_emprestimo;

    private String moeda;

    private BigDecimal valor_obtido;

    private BigDecimal valor_final;

    private LocalDateTime data_vencimento;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;
}

File: application/src/util/interfaces/loan.ts
export interface LoanForm {
    id: number;
    data_emprestimo: string;
    data_vencimento: string;
    moeda: string;
    valor_obtido: string;
    valor_final: string;
    customer_id: number
};

export interface LoanResponse {
    id: number;
    data_emprestimo: string;
    moeda: string;
    valor_obtido: number;
    data_vencimento: string;
    valor_final: number;
    customer: {
        id: number;
        cpf: string;
        name: string
    }
};